sdk/tests/conformance2/rendering/framebuffer-texture-clear.html

```diff
@@ -0,0 +1,190 @@
+<script src="../../js/webgl-test-utils.js"></script>
+<script src="../../js/tests/tex-image-and-sub-image-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+<canvas id="canvas" width="2" height="2"> </canvas>
+<script>
+"use strict";
+var wtu = WebGLTestUtils;
+var tiu = TexImageUtils;
+var gl;
+
+function testFramebufferTextureClearWithPureZeroOrOne(clearColor, expectedColor) {
+  var width = 32;
+  var height = 16;
+  var texture0 = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, texture0);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, width, height, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+  gl.bindTexture(gl.TEXTURE_2D, null);
+  var fbo0 = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo0);
+  gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture0, 0);
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo0);
+  gl.viewport(0, 0, width, height);
+  gl.disable(gl.SCISSOR_TEST);
+  gl.colorMask(true, true, true, true);
+  gl.clearColor(clearColor[0], clearColor[1], clearColor[2], clearColor[3]);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+
+  var texture1 = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, texture1);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+  gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, width, height, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
+  gl.bindTexture(gl.TEXTURE_2D, null);
+  var fbo1 = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fbo1);
+  gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.TEXTURE_2D, texture1, 0);
+
+  var vSrc =
+        "#version 300 es\n"
+      + "precision highp float;\n"
+      + "in vec2 inPosition;\n"
+      + "in vec2 inLocalCoord;\n"
+      + "out vec2 texCoord;\n"
+      + "void main() {\n"
+      + "    texCoord = inLocalCoord;\n"
+      + "    gl_Position = vec4(inPosition, 0.0, 1.0);\n"
+      + "}\n";
+
+  var fSrc =
+        "#version 300 es\n"
+      + "precision highp float;\n"
+      + "out vec4 fragColor;\n"
+      + "uniform sampler2D uTextureSampler;\n"
+      + "in vec2 texCoord;\n"
+      + "void main() {\n"
+      + "    fragColor = texture(uTextureSampler, texCoord);\n"
+      + "}\n";
+
+  var program = wtu.loadProgram(gl, vSrc, fSrc);
+  if (!program) {
+    testFailed("Failed to create the program");
+    return;
+  }
+
+  gl.bindAttribLocation(program, 0, "inPosition");
+  gl.bindAttribLocation(program, 1, "inLocalCoord");
```

Comment:
These calls have no effects until you relink the program.  So it only works because of pure luck that the binding auto-generated is the same as you wanted. Instead, you should do getAttribLocation() here.